03-pipefilter.md

```diff
@@ -16,277 +16,465 @@ Now that we know a few basic commands,
-  15  111  825 propane.pdb
- 107  819 6081 total
+$ ls *.pdb > content-of-molecules
 ~~~
 
-> ## Wildcards {.callout}
-> 
-> `*` is a **wildcard**. It matches zero or more
-> characters, so `*.pdb` matches `ethane.pdb`, `propane.pdb`, and every
-> file that ends with '.pdb'. On the other hand, `p*.pdb` only matches 
-> `pentane.pdb` and `propane.pdb`, because the 'p' at the front only 
-> matches filenames that begin with the letter 'p'.
-> 
-> `?` is also a wildcard, but it only matches a single character. This
-> means that `p?.pdb` matches `pi.pdb` or `p5.pdb`, but not `propane.pdb`.
-> We can use any number of wildcards at a time: for example, `p*.p?*`
-> matches anything that starts with a 'p' and ends with '.', 'p', and at
-> least one more character (since the '?' has to match one character, and
-> the final '\*' can match any number of characters). Thus, `p*.p?*` would
-> match `preferred.practice`, and even `p.pi` (since the first '\*' can
-> match no characters at all), but not `quality.practice` (doesn't start
-> with 'p') or `preferred.p` (there isn't at least one character after the
-> '.p').
-> 
-> When the shell sees a wildcard, it expands the wildcard to create a
-> list of matching filenames *before* running the command that was
-> asked for. As an exception, if a wildcard expression does not match
-> any file, Bash will pass the expression as a parameter to the command
-> as it is. For example typing `ls *.pdf` in the molecules directory
-> (which contains only files with names ending with `.pdb`) results in
-> an error message that there is no file called `*.pdf`.
-> However, generally commands like `wc` and `ls` see the lists of
-> file names matching these expressions, but not the wildcards
-> themselves. It is the shell, not the other programs, that deals with
-> expanding wildcards, and this another example of orthogonal design.
+The greater than symbol, `>`, tells the shell to **redirect** the command's output
+to a file, in this case `content-of-molecules` instead of printing it to the screen.
+The shell will create the file if it doesn't exist,
+or overwrite the contents of that file if it does.
+(This is why there is no screen output:
+everything that `ls` would have printed has gone into the file
+`content-of-molecules` instead.)
+
+> ## Redirecting Input {.callout}
+>
+> As well as using `>` to redirect a program's output, we can use `<` to
+> redirect its input, i.e., to read from a file instead of from standard
+> input. For example, instead of writing `wc ammonia.pdb`, we could write
+> `wc < ammonia.pdb`. In the first case, `wc` gets a command line
+> parameter telling it what file to open. In the second, `wc` doesn't have
+> any command line parameters, so it reads from standard input, but we
+> have told the shell to send the contents of `ammonia.pdb` to `wc`'s
+> standard input.
 
-If we run `wc -l` instead of just `wc`,
-the output shows only the number of lines per file:
+For (2) we will use `wc`
```

Comment:
Re-explain what (2) is, something like "now we want to be able to count the lines of `content-of-molecules`, as they will tell us how many files were in the directory"